runtime/vm/ContinuationHelpers.cpp

```diff
@@ -198,8 +223,19 @@ yieldContinuation(J9VMThread *currentThread)
 	 *
 	 * must be maintained for weakly ordered CPUs, to unsure that once the continuation is again available for GC scan (on potentially remote CPUs), all CPUs see up-to-date stack .
 	 */
-	Assert_VM_true((uintptr_t)currentThread == continuation->state);
-	continuation->state = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+	Assert_VM_true(VM_VMHelpers::isContinuationMountedWithCarrierThread(continuation->state, currentThread));
+
+	j9object_t continuationObject = J9VMJAVALANGTHREAD_CONT(currentThread, currentThread->carrierThreadObject);
+	/* Notify GC of Continuation stack swap */
+	jboolean finished = J9VMJDKINTERNALVMCONTINUATION_FINISHED(currentThread, continuationObject);
+
+	if (!finished) {
```

Comment:
pull out resetContinuationCarrierID in front of the 'if' - it's common operation for both cases

Reply:
from perspective of postUnmount call wondering if it should be reset after....
then, it would be a mirror to what we do during mount (first call sync, what sets carrierID, and then do preMount)

Reply:
```
	if (finished) {
		VM_VMHelpers::setContinuationFinished(continuation);
	}
	VM_VMHelpers::resetContinuationCarrierID(continuation);
	if (!finished) {
		currentThread->javaVM->memoryManagerFunctions->postUnmountContinuation(currentThread, continuationObject);
	}

```